cli/interface.go

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) 2018 Huawei Corporation.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+package main
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/urfave/cli"
+	"text/tabwriter"
+
+	vc "github.com/kata-containers/runtime/virtcontainers"
+)
+
+var interfaceCLICommand = cli.Command{
+	Name:  "interface",
+	Usage: "manage interfaces for container",
+	Subcommands: []cli.Command{
+		infAddCommand,
+		infListCommand,
+		infUpdateCommand,
+	},
+	Action: func(context *cli.Context) error {
+		return cli.ShowSubcommandHelp(context)
+	},
+}
+
+var infAddCommand = cli.Command{
+	Name:      "add",
+	Usage:     "add a new nic into a container",
+	ArgsUsage: "<container-id> <interface-name>",
+	Flags:     []cli.Flag{},
+	Action: func(context *cli.Context) error {
+		if context.Args().Present() == false {
+			return fmt.Errorf("missing container ID, should provide one")
+		}
+		containerID := context.Args().First()
+		status, sandboxID, err := getExistingContainerInfo(containerID)
+		if err != nil {
+			return err
+		}
+
+		containerID = status.ID
+		// container MUST be running
+		if status.State.State != vc.StateRunning {
+			return fmt.Errorf("container %s is not running", containerID)
+		}
+		ifName := context.Args().Get(1)
+		if ifName == "" {
+			return fmt.Errorf("interface name should not be empty")
+		}
+		return vci.AddNetwork(sandboxID, ifName)
+	},
+}
+
+var infListCommand = cli.Command{
+	Name:      "ls",
+	Aliases:   []string{"list"},
+	Usage:     "list network interfaces in a container",
+	ArgsUsage: `ls <container-id>`,
+	Flags:     []cli.Flag{},
+	Action: func(context *cli.Context) error {
+
+		tw := tabwriter.NewWriter(os.Stdout, 10, 1, 3, ' ', 0)
+		fmt.Fprintln(tw, "Name\tMac\tIP\tMtu")
+
+		tw.Flush()
+		return nil
+	},
+}
+
+var infUpdateCommand = cli.Command{
+	Name:      "update",
+	Usage:     "update configuration of interface",
+	ArgsUsage: `update <container-id> <interface-name>`,
```

Comment:
Same question here, do we expect to provide a simple interface name or a full description ?